### DISCUSSION TITLE
Airplay Group - Song restarts when turning on a group member - works fine for turning off

### DISCUSSION THREAD

**Discussion by bbruenings** [music-assistant/support#2870] [Q&A] [answered]
Hey together,

first of all great work on Music Assistant. It is a great Add-On and I'm really looking forward using it in my Home Assistant setup.

I was wondering if anyone else has the same issue regarding Airplay Groups.

Every time a speaker is turned on in a existing group (or added to an airplay speaker that plays for itself) the current song get's restarted.

I tried it with Sonos Speakers over Airplay and HomePods. As I have a mixture of Sonos and HomePods I would love to control the music through Music Assistant using the Airplay Provider.

As of right now using all Speakers in a Group and changing just the volume would be my workaround. 

Thanks for any hints.

Keep up the great work!

Bennet

**Comment by OzGav** [comment]
@marcelveldt thoughts?

**Comment by marcelveldt** [comment]
If you add a speaker to a syncgroup (either by using a syncgroup or manually syncing it) we need to RESTART the stream so it will be quickly (0.5 seconds) stopped and then restarts but it should resume exactly where it left off.

We did however have a generic issue with resuming which is fixed in beta but not yet in stable - we'll publish a new stable version out soon

**Reply by bbruenings** [reply]
Thank you for the quick reply. Unfortunately it also happens on the new Beta. Is there any good way to check the logs for it? The Add-On log only states that it starts the Queue Flow stream to the master speaker. That is when I first start playback on the master speaker and also when adding the second speaker.
![image](https://github.com/user-attachments/assets/d7a23c2f-45b2-401d-9046-6e2a0d51cbc7)

**Reply by marcelveldt** [reply]
Can you describe more exactly WHAT happens ? I'm having a hard time following your exact issue. You did read my comment above that it is correct that we restart a stream when you add an additional player ?

**Reply by bbruenings** [reply]
Sorry for that.

The setup is using the Airplay Provider for all the Speakers

- I select one speaker and play a song on it
- After 10 or 15 seconds I link a second speaker

**Expected behavior**: Restart the stream on the same position of the song
**Current behavior**: Restart the stream at the beginning of the song

The playback details (at the bottom of the page) show the correct progress - but the song is started from the beginning.

I hope that helps - let me know if you need any other information.

**Reply by OzGav** [reply]
How are you linking the second player?

**Reply by bbruenings** [reply]
I tried two different approaches. 

**First approach**
Starting on a single speaker and linking by clicking on the link button of that speaker and adding a second speaker
![image](https://github.com/user-attachments/assets/4925cb55-162b-4c3b-bea5-2d3171d76a2a)
 
**Second approach**

- Creating an airplay group with multiple players
- Starting playback for that group
- Turning on a second player of that group
![image](https://github.com/user-attachments/assets/72b9e2e4-2970-48af-b2ba-69de5fd06eeb)

In either scenario after turning on/linking the second speaker the song starts from the beginning. The progress bar shows the correct progress but the stream seems to start the song from the beginning.

**Reply by OzGav** [reply]
Groups are being reworked at the moment so keep an eye on the updates

**Reply by bbruenings** [reply]
Thanks - I will be watching :-)

**Reply by OzGav** [reply, marked as answer]
beta 30 is released

**Reply by bbruenings** [reply]
Thanks :-) I updated it this morning and just tried it. Seems to work now :-) at least with Spotify. Need to do another test with Apple Music when I refreshed the token.

Keep up the great work and thanks for supporting!
